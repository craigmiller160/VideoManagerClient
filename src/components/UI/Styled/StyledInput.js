import styled from 'styled-components';

const StyledInput = styled.input`
    border: 1px solid lightgray;
    border-radius: 3px;
    display: block;
    width: 100%;
    font-size: 1rem;
    padding: 2px 8px;
    line-height: 1.5;
    font-weight: 400;
    height: calc(2.25rem + 2px);
`;
StyledInput.displayName = 'StyledInput';

export default StyledInput;