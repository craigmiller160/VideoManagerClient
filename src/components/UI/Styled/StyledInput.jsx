/*
 *     VideoManagerClient
 *     Copyright (C) 2020 Craig Miller
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledInput = styled.input`
	border: 1px solid ${(props) => (props.hasError ? '#dc3545' : 'lightgray')};
	border-radius: 3px;
	display: block;
	width: 100%;
	font-size: 1rem;
	padding: 2px 8px;
	line-height: 1.5;
	font-weight: 400;
	height: calc(2.25rem + 2px);
`;
StyledInput.displayName = 'StyledInput';
StyledInput.propTypes = {
	hasError: PropTypes.bool
};
StyledInput.defaultProps = {
	hasError: false
};

export default StyledInput;
