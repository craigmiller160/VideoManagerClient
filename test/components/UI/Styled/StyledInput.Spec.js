import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components'
import StyledInput from 'components/UI/Styled/StyledInput';

describe('StyledInput', () => {
    it('renders correctly', () => {
        const component = mount(<StyledInput/>);
        expect(component).toMatchSnapshot();
    });
});