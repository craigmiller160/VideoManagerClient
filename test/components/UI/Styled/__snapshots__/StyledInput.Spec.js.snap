// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput renders correctly 1`] = `
.c0 {
  border: 1px solid lightgray;
  border-radius: 3px;
  display: block;
  width: 100%;
  font-size: 1rem;
  padding: 2px 8px;
  line-height: 1.5;
  font-weight: 400;
  height: calc(2.25rem + 2px);
}

<StyledInput>
  <input
    className="c0"
  />
</StyledInput>
`;
